test/integ/test_cursor.py

```diff
@@ -744,11 +745,9 @@ def test_closed_cursor(conn, db_parameters):
         c.close()
 
         fmt = "select aa from {name}".format(name=db_parameters["name"])
-        try:
+        with pytest.raises(InterfaceError, match="Cursor is closed in execute") as err:
             c.execute(fmt)
-            raise Exception("should fail as the cursor was closed.")
-        except snowflake.connector.Error as err:
-            assert err.errno == errorcode.ER_CURSOR_IS_CLOSED
+        assert err.errno == errorcode.ER_CURSOR_IS_CLOSED
```

Comment:
Test failed here with `AttributeError: 'ExceptionInfo' object has no attribute 'errno'`. Not sure how to fix it offhand.

Reply:
I fixed it - we needed to retrieve err.value first.